### ISSUE TITLE
Endless rows of January 1st

### ISSUE THREAD

**Issue by DestructionOfPlanetJupiter** [Meecast/meecast#31]
There still appears to be something off with the switch from one year to the next. I have a Sailfish one and since I have updated the stations a couple of days ago, I have an endless(?) row of January 1st entries. At least there are entries now unlike in previous years, but there appears to be still an error in your app. I can provide a screenshot if necessary. Also the app takes quite a while to start with these amount of entries. 

The app version is 1.1.20. I can provide a screenshot if necessary.

**Comment by Olf0** [comment]
Well, I have the same bug as every turn of the year: No data downloaded for dates after the 31st of December, until the 1st of January.

@DestructionOfPlanetJupiter, supposedly your new station settings made a difference here.

@vasvlad, I also hoped for this bug being fixed, as it was originally reported years ago, IIRC.  
But obviously nobody (including me) took the effort to file it here (at least I have not found it).

**Comment by DestructionOfPlanetJupiter** [comment]
@Olf0 Yes, I had the problem you have outlined last year as well. 
The app runs on a Jolla 1 and on Sailfish 3.0.0.8.

**Comment by DestructionOfPlanetJupiter** [comment]
Update in the new year provides empty data rows. 

... great ...

**Comment by Olf0** [comment]
@DestructionOfPlanetJupiter,

> Update in the new year provides empty data rows.

No, at least not for me and as every year: Weather data was completely updated (7 days forecast) by the first scheduled update in the new year.

> edit:
New enties reveal proper data sets ..., so all had to be added again. Which is rather unconvenient. Furthermore, I do not know the weather stations, so I have to set up all again anew ... should a further update on the 2nd January should proof to be insufficient

I guess you just should have been patient.

**Comment by vasvlad** [comment]
Fixed

**Comment by DestructionOfPlanetJupiter** [comment]
Problem still exists. Version 1.1.23

**Comment by vasvlad** [comment]
> Problem still exists. Version 1.1.23

Try version 1.1.24
https://openrepos.net/content/vasvlad/meecast-sailfishos

**Comment by Olf0** [comment]
@vasvlad, rereading this I realise we are bad bug reporters.
As this can only be observed on the last six days before each turn of the year, my intention is to create at least a proper bug report this year.

My testing environment:
* MeeCast 1.1.23-8 under SailfishOS 2.2.1 on a Jolla 1
* MeeCast 1.1.24-1 under SailfishOS 3.2.1 on an Xperia X

Both have weather data for the same three locations subscribed from Weather.com (only).

Side note:
IIRC, earlier (i.e., years ago), MeeCast did show **N/A** (not available) when Weather.com was used as data provider, in the Events View (for the currently selected location) and each location main page in MeeCast for any date after December 31.  At the turn of the year (i.e., on January 1st) this "automatically resolved itself" only to reappear again on the next December 25 (because it is 7 days before January 1st).
This is definitely fixed (AFAIR at the beginning of 2018), i.e. the weather data from Weather.com is correctly retrieved now for dates in the next year.
<br />
What I observe now (since retrieving the data from Weather.com has been fixed) is merely an issue how MeeCast displays the weather data for the last day of the seven days forecast in each location main page:
The line with the information for the last day of the forecast is displayed 23 times, followed by approximately the same amount (~ 23) of blank lines.

Screenshots of this issue:
* MeeCast 1.1.23-8 under SailfishOS 2.2.1 on a Jolla 1 in portrait orientation
   * The first six days are displayed fine
      ![Screenshot_20191231_001](https://user-images.githubusercontent.com/16547876/71626922-ed802d80-2bef-11ea-994d-3208128b1865.png)
   * The last day is displayed 23 times
      ![Screenshot_20191231_002](https://user-images.githubusercontent.com/16547876/71626936-04bf1b00-2bf0-11ea-8f47-2c365d239341.png)
   * Followed by many blank lines
      ![Screenshot_20191231_003](https://user-images.githubusercontent.com/16547876/71626955-1a344500-2bf0-11ea-8284-9b03c7b49b4b.png)
      ![Screenshot_20191231_004](https://user-images.githubusercontent.com/16547876/71626966-24564380-2bf0-11ea-92f1-5d298f9a3645.png)
* MeeCast 1.1.24-1 under SailfishOS 3.2.1 on an Xperia X in landscape orientation
   * The first six days are displayed fine
   * The last day is displayed 23 times
   * Followed by many blank lines

Hence this issue still exists in MeeCast 1.1.24-1!
Please reopen.

**Comment by DestructionOfPlanetJupiter** [comment]
harbour-meecast-1.1.24-1.armv7hl.rpm has another issue, the version number in Info is wrong. It says: 1.1.22, which is not something we have installed. Maybe you fix this thing along the way.
